core/src/test/java/org/apache/druid/java/util/common/BytesTest.java

```diff
@@ -0,0 +1,300 @@
+
+    try {
+      Bytes.parse("1 b", 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+      Assert.assertTrue(true);
+    }
+
+    try {
+      Bytes.parse("tb", 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+      Assert.assertTrue(true);
+    }
+
+    try {
+      Bytes.parse("1 mb", 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+      Assert.assertTrue(true);
+    }
+
+    try {
+      Bytes.parse("gb", 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+      Assert.assertTrue(true);
+    }
+
+    try {
+      Bytes.parse("tb", 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+      Assert.assertTrue(true);
+    }
+  }
+
+  @Test
+  public void testOverflow()
+  {
+    try {
+      Bytes.parse("123456789123456789123", 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / 1000 + 1) + "k";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / 1000_000 + 1) + "m";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / 1000_000_000L + 1) + "g";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / 1000_000_000_000L + 1) + "t";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / 1024 + 1) + "kb";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / (1024 * 1024) + 1) + "mb";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / (1024L * 1024 * 1024) + 1) + "gb";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+
+    try {
+      String max = (Long.MAX_VALUE / (1024L * 1024 * 1024 * 1024) + 1) + "tb";
+      Bytes.parse(max, 0);
+      Assert.assertFalse("IAE should be thrown", true);
+    }
+    catch (IAE e) {
+    }
+  }
+
+  @Test
+  public void testJSON() throws JsonProcessingException
+  {
+    ObjectMapper mapper = new ObjectMapper();
+    Bytes bytes = new Bytes("5m");
+    String serialized = mapper.writeValueAsString(bytes);
+    Bytes deserialized = mapper.readValue(serialized, Bytes.class);
+    Assert.assertEquals(bytes, deserialized);
+  }
+
+  public static class BytesRangeTest
```

Comment:
We usually name a test class to have the `Test` suffix if it has unit tests. How about `TestBytesRange`?